### ISSUE TITLE
Have a function to see if device is already registered.

### ISSUE THREAD

**Issue by roberto-robles** [AppGyver/PushNotifications#14]
Is it possible to have this feature for this plugin, this could make things all validations easier on our side.

**Comment by harsaharsa** [comment]
Valid point, I'll poke around and see how it would be doable. If you find any reference material online on some best practices on the native side, please link!

**Comment by roberto-robles** [comment]
Thanks @harsaharsa , I'll take a look into that, and if I find something, I'll share with you guys shortly.

**Comment by fidoboy** [comment]
AFAIK this has no sense because the logical thing is that the devices DB is located on your server. The client side only sends the token to the server and then the server sends its response saying if device is already on DB or not. Can't see how this plugin could replace or enhance this process in other way than it's right now... But may be I'm wrong. Just my 2c

**Comment by roberto-robles** [comment]
@fidoboy you are right when you say that device push token should be stored in a DB backend in a server, but I am talking to the case that app that has multiple Users. We came to the solution of storing device token in LocalStorage, but if User register the app and then uninstall the app and install it again when our flow "tries" to register again the device, it won't work cause the device is already registered (by the way this only matters in Android because iOS implementation always returns the same token and success callback always works, regardless of the state of the app (register or not with APNS).

So if what I am asking is possible it will be so much easier to handle all this corner cases. One other posibility is a way to detect if user uninstalls the app, so before that we could unregister the phone (I know this is almost imposible)

**Comment by harsaharsa** [comment]
As you said, on iOS it's enough to just call the register function every time.

For Android, it looks like the best practice would be to implement something like this. See http://developer.android.com/google/gcm/client.html – there's a sample implementation of `getRegistrationId`. Pull requests are welcome! :)
